http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -703,26 +727,39 @@ public void onContent(Response response, ByteBuffer content)
         // process response
         long responseStart = System.nanoTime();
 
-        JettyResponse jettyResponse = null;
         T value;
-        try {
-            jettyResponse = new JettyResponse(response, listener.getInputStream());
-            value = responseHandler.handle(request, jettyResponse);
+
+        if (isStreaming) {
+            JettyResponse jettyResponse;
+            try {
+                jettyResponse = new JettyResponse(response, listener.getInputStream());
+            }
+            catch (Throwable e) {
+                throw propagate(request, e);
+            }
+
+            Runnable finalizer = () -> finalizeExecution(request, span, requestStart, requestSize, responseStart, jettyResponse);
+            try {
+                value = responseHandler.handle(request, new JettyStreamingResponse(jettyResponse, span, finalizer));
+            }
+            catch (RuntimeException e) {
+                throw new StreamingException(e, finalizer);
```

Comment:
why pass `finalizer` through `StreamingException` instead calling it immediately here before exception is thrown?

Reply:
So that `span.end()` is the very last thing called. As I read the code this needs to be the last operation. For example, `finalizeExecution()` sets an attribute on the span. It could be that I don't understand the span library well enough.

Reply:
So that `span.end()` is the very last thing called. As I read the code this needs to be the last operation. For example, the catch in `execute` calls `span.setStatus()` and that should happen before `span.end()`.